service/scaladsl/server/src/main/scala/com/lightbend/lagom/scaladsl/server/LagomServer.scala

```diff
@@ -56,7 +56,7 @@ trait LagomServerComponents {
 
   lazy val lagomServerBuilder: LagomServerBuilder = new LagomServerBuilder(httpConfiguration, serviceResolver)(materializer, executionContext)
   protected def bindService[T <: Service]: LagomServiceBinder[T] = macro ScaladslServerMacroImpl.createBinder[T]
-  protected def serverFor[T <: Service]: (T) => LagomServer = macro ScaladslServerMacroImpl.simpleBind[T]
+  protected def serverFor[T <: Service](serviceFactory: => T): LagomServer = macro ScaladslServerMacroImpl.simpleBind[T]
```

Comment:
When creating the binding, the implementation must be a factory (aka call-by-name argument). Fortunately it is possible (since scala 2.11.0) to implement "call-by-name with scala macros"(https://issues.scala-lang.org/browse/SI-5778). Without out that feature all the rewrite of the scaladsl would collapse (as demonstrated in the last Travis build were 4/5 tasks failed).